src/lib/core/fiber_channel.h

```diff
@@ -207,6 +215,12 @@ fiber_channel_is_empty(struct fiber_channel *ch)
 	return ch->count == 0;
 }
 
+/**
+ * Setter for fiber_channel_graceful_close.
+ */
+void
+fiber_channel_set_graceful_close(bool is_graceful);
```

Comment:
I thought on it once again and it seems I found how to naturally represent this duality.

```c
/** The fiber channel has two ways to end its life. */
enum fiber_channel_close_mode {
	/**
	 * The forceful close mode means that fiber_channel_close()
	 * discards all stored messages, wake ups all readers and
	 * writers and forbids to read or write messages. The
	 * channel is effectively died after closing.
	 *
	 * fiber_channel_close() works exactly as
	 * fiber_channel_destroy() when this mode is in effect.
	 *
	 * This mode represents original behavior of the fiber
	 * channel since its born and hence it is the default.
	 */
	FIBER_CHANNEL_CLOSE_FORCEFUL,
	/**
	 * The graceful close mode means that fiber_channel_close()
	 * only marks the channel as read-only: no new messages can
	 * be added, but all buffered messages can be received.
	 * All waiting writers receive an error.
	 *
	 * A closed channel is automatically destroyed when all
	 * buffered messages are read.
	 *
	 * This mode is considered as more safe and recommended for
	 * users. The main idea behind it is to prevent accidental
	 * data loss or, in other words, fit users' expectation that
	 * all accepted messages will be delivered. This reflects
	 * how channels work in other programming languages, for
	 * example, in Go.
	 */
	FIBER_CHANNEL_CLOSE_GRACEFUL,
};

/**
 * Choose how the channels end its life.
 *
 * Affects all channels.
 *
 * The behavior is unspecified for already created channels.
 * Choose the mode at an early stage of application's
 * initialization.
 *
 * @sa enum fiber_channel_close_mode
 * @sa fiber_channel_close
 */
void
fiber_channel_set_close_mode(enum fiber_channel_close_mode);
```

All comments are to illustrate the idea. Feel free to modify them to make better.

(I would highlight the unspecified behavior note. It should be declared explicitly.)

Reply:
I'm not a big fan of `enum` with just two values. Maybe keep `bool` but add more comments?

Reply:
Igor applied this proposal. Closing the discussion.